### PR TITLE
Adding blur processing to image processor

### PR DESCRIPTION
**Pull request by rubish**
- Adding blur processing
- Extracting extend as different segment
- Fixing some non-erroneous cases logged as errors
- Better explanation of parameters in readme

### CODE REVIEW

**Comment by rubish** [comment]
@vikalpj Can you take a look and merge this. After this change we have also added autorotate option which is enabled by default and can be disabled. Creating a pull request for autorotate is pulling in all these changes too.

**Comment by vikalpj** [comment]
@rubish Seems like I missed this, will review this and merge asap.

File: app/paramParser.js [hunk 1]
```diff
@@ -29,6 +35,50 @@ function parseProcessor(key) {
...
+function parseAutoRotate(key){
+    var splitArray = key.split('/');
+    var regex = /auto_rotate:(true|false)/;
+    var regexMatch = splitArray[0].match(regex);
+    var auto_rotate = true;
+
+    if (regexMatch && regexMatch.length > 0) {
+        splitArray.splice(0, 1);
+        var rotate = JSON.parse(regexMatch[1]);
+        if (rotate == false) {
```
Comment (vikalpj) [inline app/paramParser.js]: this can just be (!rotate)

Comment (vikalpj) [reply app/paramParser.js]: or even 
just a single line...

auto_rotate = !JSON.parse(regexMatch[1]) if value is undefined as well then also will set this to true.. 

Also do you think JSON.parse ? is a good approach... can we just not do

auto_rotate = !(regexMatch[1].toLowerCase() == "false") or something like this.

File: app/processImage.js [hunk 2]
```diff
@@ -41,14 +45,18 @@ function processImage(key, imageParams, processImageCallback) {
...
         function(data, fileInfo, callback) {
             // save file to S3
             console.log("Saving file to storage");
             storage.storage.saveFile(key.replace('/',''), data, fileInfo, callback);
-        }
+        },
     ], function(err, data) {
-        // this function is always executed both in case of err and success as well
-        console.log("Error raised while Processing raw File, Error=", err, ", data=", data);
+        if(err) {
```
Comment (vikalpj) [inline app/processImage.js]: Is there a reason we wrapped this in error. We might not be able to identify this whole block was called or not.

This whole block is the error block  for `waterfall`

Comment (harshal27) [reply app/processImage.js]: Its the last callback to be executed once all the functions of waterfall have been completed.
The block is wrapped in error as It should only log when there is actually some king of error.

Comment (vikalpj) [reply app/processImage.js]: gotcha. Thanks for the explaination 👍

**Comment by vikalpj** [comment]
Merging this... 👍 💯

**Pull request merged**